tests/GoodsTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+use PHPUnit\Framework\TestCase;
+use Stock\Models\Goods;
+
+class GoodsTest extends TestCase
+{
+    public $goods_data = [];
+
+    public function setUp()
+    {
+        $this->goods_data = ['name' => 'brakes', 'price' => 199.00, 'producer' => 'acme', 'quantity' => 20];
+    }
+
+
+    public function testCheckTypeMethod()
+    {
+        $goods = new Goods($this->goods_data);
+        
+        $class = new ReflectionClass($goods);
+        $method = $class->getMethod('check_type');
+        $method->setAccessible(true);
+
+        $this->assertTrue($method->invokeArgs($goods, ['silnik', 'string']));
+        $this->assertTrue($method->invokeArgs($goods, [124, 'integer']));
+        $this->assertTrue($method->invokeArgs($goods, [199.00, 'double']));
+        $this->assertFalse($method->invokeArgs($goods, [20, 'string']));
+        $this->assertFalse($method->invokeArgs($goods, [199, 'double']));
+    }
+
+    public function testCheckRequiredMethod()
+    {
+        $goods = new Goods($this->goods_data);
+        
+        $class = new ReflectionClass($goods);
+        $method = $class->getMethod('check_required');
+        $method->setAccessible(true);
+
+        $this->assertTrue($method->invokeArgs($goods, ['silnik', true]));
+        $this->assertTrue($method->invokeArgs($goods, [ null, false]));
+        $this->assertFalse($method->invokeArgs($goods, [ null, true]));
+    }
+
+    public function testGoodsValidationMethod()
+    {
+        $goods = new Goods($this->goods_data);
```

Comment:
To samo co wyżej.
Nie wrzucamy nic do konstruktora, skoro później budujemy i tak nowe tożsamości za każdym razem przy aż 3 asercjach.